p2p/server/server.go

```diff
@@ -188,18 +294,21 @@ func (s *Server) Request(ctx context.Context, pid peer.ID, req []byte, resp func
 			failure(err)
 			return
 		}
-		if err := wr.Flush(); err != nil {
+		err = wr.Flush()
+		if err != nil {
```

Comment:
I don't understand this and the change below? The behaviour is still the same as before?

Reply:
the change is to make `err` visible in defer. the behavior is the same

Reply:
I feel like the code would be easier to follow if instead of the `defer` the body of the `go func` would be its own method:

```go
go func() {
	err := s.streamData(...)
	s.logger.WithContext(ctx).With().Debug("request execution time",
		log.String("protocol", s.protocol),
		log.Duration("duration", time.Since(start)),
		log.Err(err),
	)
	s.trackClientLatency(start, err)
}()
```

Reply:
yes you are right, i will change